Source/WebCore/platform/graphics/texmap/coordinated/CoordinatedPlatformLayer.cpp

```diff
@@ -908,8 +912,11 @@ void CoordinatedPlatformLayer::flushCompositingState(TextureMapper& textureMappe
         layer.setSolidColor(m_contentsColor);
 
 #if ENABLE(DAMAGE_TRACKING)
-    if (m_pendingChanges.contains(Change::Damage))
-        layer.setDamage(WTFMove(m_damage));
+    if (m_pendingChanges.contains(Change::Damage)) {
+        ASSERT(m_damage.has_value());
+        layer.setDamage(WTFMove(*m_damage));
+        m_damage = std::nullopt;
```

Comment:
I was recently reading the _Safer C++ Guidelines_, where it says  [Use std::exchange() instead of WTFMove() when the "moved-from" variable may get reused](https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines#use-stdexchange-instead-of-wtfmove-when-the-moved-from-variable-may-get-reused)

If we followed that to the letter, we could write the above as 

```
        layer.setDamage(WTFMove(*std::exchange(m_damage, std::nullopt)));
```

but I believe the code as it stands is more readable